constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/SpansContains.java

```diff
@@ -0,0 +1,121 @@
+package gov.nasa.jpl.aerie.constraints.tree;
+
+import gov.nasa.jpl.aerie.constraints.model.EvaluationEnvironment;
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.Spans;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.StreamSupport;
+
+
+public record SpansContains(Expression<Spans> parents, Expression<Spans> children, Requirement requirement) implements Expression<Windows> {
+  public record Requirement(
+      Optional<Integer> minCount,
+      Optional<Integer> maxCount,
+      Optional<Expression<Duration>> minDur,
+      Optional<Expression<Duration>> maxDur
+  ) {
+    public static SpansContains.Requirement newDefault() {
+      return new Requirement(
+          Optional.of(1),
+          Optional.empty(),
+          Optional.empty(),
+          Optional.empty()
+      );
+    }
+  }
+
+  @Override
+  public Windows evaluate(final SimulationResults results, final Interval bounds, final EvaluationEnvironment environment) {
+    final var parents = this.parents.evaluate(results, bounds, environment);
+    final var children = this.children.evaluate(results, bounds, environment);
+    var falseIntervals = new ArrayList<Interval>();
+
+    // Check for count requirements if they exist
+    if (this.requirement.minCount.isPresent() || this.requirement.maxCount.isPresent()) {
+      final var sortedParents = StreamSupport.stream(parents.spliterator(), true).sorted((l, r) -> l
+          .interval()
+          .compareStarts(r.interval())).toList();
+      final var sortedChildren = StreamSupport.stream(children.spliterator(), true).sorted((l, r) -> l
+          .interval()
+          .compareStarts(r.interval())).toList();
+      var childIndex = 0;
+
+      for (final var parent: sortedParents) {
+        final var parentInterval = parent.interval();
+        var childInterval = sortedChildren.get(childIndex).interval();
+        while (childInterval.compareStarts(parentInterval) == -1) {
+          childIndex++;
+          childInterval = sortedChildren.get(childIndex).interval();
+        }
+
+        var instanceCount = new AtomicInteger(0);
+        var transientChildIndex = childIndex;
+        while (parentInterval.contains(childInterval)) {
+          instanceCount.getAndIncrement();
+          transientChildIndex++;
+          childInterval = sortedChildren.get(transientChildIndex).interval();
```

Comment:
In my attempt at a manual test I ran into an index out of bounds error here

```typescript
export default () => {
  return Spans.ForEachActivity(ActivityType.GrowBanana).contains(Spans.ForEachActivity(ActivityType.BiteBanana));
}
```


![Screenshot 2023-10-18 at 11 50 10 AM](https://github.com/NASA-AMMOS/aerie/assets/1189602/b1f32023-38fc-4834-8bc2-d7b011f9bc04)

```
aerie_merlin  | java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
aerie_merlin  |         at java.base/java.util.ImmutableCollections$ListN.get(Unknown Source)
aerie_merlin  |         at gov.nasa.jpl.aerie.constraints.tree.SpansContains.evaluate(SpansContains.java:63)
aerie_merlin  |         at gov.nasa.jpl.aerie.constraints.tree.SpansContains.evaluate(SpansContains.java:17)
aerie_merlin  |         at gov.nasa.jpl.aerie.constraints.tree.ViolationsOfWindows.evaluate(ViolationsOfWindows.java:25)
aerie_merlin  |         at gov.nasa.jpl.aerie.constraints.tree.ViolationsOfWindows.evaluate(ViolationsOfWindows.java:16)
aerie_merlin  |         at gov.nasa.jpl.aerie.constraints.tree.Expression.evaluate(Expression.java:18)
aerie_merlin  |         at gov.nasa.jpl.aerie.merlin.server.services.ConstraintAction.getViolations(ConstraintAction.java:209)
aerie_merlin  |         at gov.nasa.jpl.aerie.merlin.server.http.MerlinBindings.getConstraintViolations(MerlinBindings.java:239)
```

Reply:
Thanks, fixed. (and added a test)